var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -80,32 +80,56 @@ def install(self, spec, prefix):
             optflags = {
                 'gcc': ['-O2',
                         '-ffast-math',
-                        '-ffree-form',
-                        '-ffree-line-length-none',
                         '-ftree-vectorize',
                         '-funroll-loops',
                         '-mtune=native'],
                 'intel': ['-O2',
                           '-pc64',
-                          '-unroll',
-                          '-heap-arrays 64']
+                          '-unroll']
             }
+
+            dflags = ['-DNDEBUG']
+
             cppflags = [
                 '-D__FFTW3',
                 '-D__LIBINT',
                 spec['fftw'].headers.cpp_flags
             ]
+
+            if '^mpi@3:' in spec:
+                cppflags.append('-D__MPI_VERSION=3')
+            elif '^mpi@2:' in spec:
+                cppflags.append('-D__MPI_VERSION=2')
+
+            if '^intel-mkl' in spec:
+                cppflags.append('-D__FFTSG')
+
+            cflags = copy.deepcopy(optflags[self.spec.compiler.name])
+            cxxflags = copy.deepcopy(optflags[self.spec.compiler.name])
             fcflags = copy.deepcopy(optflags[self.spec.compiler.name])
-            fcflags.append(spec['fftw'].headers.cpp_flags)
+            fcflags.extend([
+                '-ffree-form',
+                '-ffree-line-length-none',
+                spec['fftw'].headers.cpp_flags
+            ])
+
+            if '%intel' in spec:
+                cflags.append('-fp-model precise')
+                cxxflags.append('-fp-model precise')
+                fcflags.extend(['-fp-model source', '-heap-arrays 64'])
```

Comment:
Is there a particular reason why the f flags are set to -fp model source instead of -fp model precise like the c and cxx flags?